### PR TITLE
On failed connection errors, log less aggressively (by an exponential…

### PR DESCRIPTION
**Pull request by mereba**
… factor) to minimize the risk of filling up disk as a result of profuse logging.

### CODE REVIEW

**Comment by mereba** [comment]
Thanks for a great library!

**Comment by videlalvaro** [comment]
Would you mind bringing your branch up to date with master?

**Comment by videlalvaro** [comment]
Also it would be nice to have an example added to the demo folder showing how to use this new feature

**Comment by mereba** [comment]
@videlalvaro,
I brought my branch up to date with master and got some failing tests :( I will take a closer look and let you know what I find.

Again, thanks for a great library!

File: PhpAmqpLib/Wire/IO/StreamIO.php [hunk 1]
```diff
@@ -103,6 +107,48 @@ public function connect()
     {
         $errstr = $errno = null;
 
+        do {
+            if ($this->context) {
```
Comment (videlalvaro) [inline PhpAmqpLib/Wire/IO/StreamIO.php]: I don't understand, is this code duplicating code from 

here: https://github.com/videlalvaro/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php#L106
here: https://github.com/videlalvaro/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php#L115

and ignoring the code from here? https://github.com/videlalvaro/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php#L85

**Comment by postalservice14** [comment]
Are you able to update your PR to pass the tests?  Also, what problem is this trying to solve?  It's a little unclear to me.  Thanks!

**Comment by postalservice14** [comment]
Seems like this PR is abandoned.  Please re-open if you'd like to bring it up to date.

**Pull request closed without merging**